code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -71,8 +71,8 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	for(var/id in cached_gases)
 		var/gas_data = cached_gases[id]
 		. += gas_data[MOLES] * gas_data[GAS_META][META_GAS_SPECIFIC_HEAT]
-	if(!.) //if no heat capacity, we're a vacuum - things get weird like this but this hack sorta works
-		. += HEAT_CAPACITY_VACUUM
+	if(!. && temperature) //if temp isn't defined, this mixture isn't a vacuum like space is. it hasn't been filled with *anything* so we want it to take on the properties of whatever gas enters it
+		. += HEAT_CAPACITY_VACUUM //however, if temp is defined but HC is still 0, then we want the mixture to behave like space does, as a heat sink

```

Comment:
thankfully, I know that if `temperature` is 0 in any mixture, it is (or should be) brand new. knowing that, I can just check to see if temperature is non-zero to give vacuums the heat capacity of space; otherwise, heat capacity will be 0 and we get expected behavior for both cases